appyters/Gene_Expression_T2D_Signatures/Gene_Expression_T2D_Signatures.ipynb

```diff
@@ -289,32 +288,44 @@
-    "    up_comb_df = comb_df[comb_df['fc'] > 0].sort_values(by='pval', ascending=True)[:10]\n",
-    "    up_comb_df = up_comb_df.drop(columns='logpv')\n",
-    "    up_comb_df['inst'] = up_comb_df.apply(lambda row: appyter_link(row.sig, row.inst), axis=1)\n",
-    "    up_comb_df['sig'] = up_comb_df.apply(lambda row: f\"* {row.sig}\" if row.pval < 0.05 else row.sig, axis=1)\n",
-    "    up_comb_df = up_comb_df.rename(columns={'sig': 'Signature', 'pval': 'P-value', 'fc': 'Log2 Fold Change', 'inst': 'Link to Bulk RNA-seq Analysis'})\n",
-    "    up_comb_df['Link to GEO Study'] = up_comb_df['Signature'].apply(geo_link)\n",
-    "\n",
-    "    dn_comb_df = comb_df[comb_df['fc'] < 0].sort_values(by='pval', ascending=True)[:10]\n",
-    "    dn_comb_df = dn_comb_df.drop(columns='logpv')\n",
-    "    dn_comb_df['inst'] = dn_comb_df.apply(lambda row: appyter_link(row.sig, row.inst), axis=1)\n",
-    "    dn_comb_df['sig'] = dn_comb_df.apply(lambda row: f\"* {row.sig}\" if row.pval < 0.05 else row.sig, axis=1)\n",
-    "    dn_comb_df = dn_comb_df.rename(columns={'sig': 'Signature', 'pval': 'P-value', 'fc': 'Log2 Fold Change', 'inst': 'Link to Analysis'})\n",
-    "    dn_comb_df['Link to GEO Study'] = dn_comb_df['Signature'].apply(geo_link)\n",
+    "def make_tables(comb_df, is_upreg):\n",
+    "    dir_df = comb_df[comb_df['fc'] > 0] if is_upreg else comb_df[comb_df['fc'] < 0]\n",
+    "    dir_df = dir_df.drop(columns='logpv').sort_values(by='pval', ascending=True)\n",
+    "    dir_df['inst'] = dir_df.apply(lambda row: appyter_link(row.sig, clickable=True, inst=row.inst), axis=1)\n",
+    "    dir_df['sig'] = dir_df.apply(lambda row: f\"* {row.sig}\" if row.pval < 0.05 else row.sig, axis=1)\n",
+    "    dir_df['pval'] = dir_df['pval'].apply(lambda x: f'{x:.3e}')\n",
+    "    dir_df['fc'] = dir_df['fc'].apply(lambda x: f'{x:.4f}')\n",
+    "    dir_df = dir_df.rename(columns={'sig': 'Signature', 'pval': 'P-value', 'fc': 'Log2 Fold Change', 'inst': 'Link to Bulk RNA-seq Analysis'})\n",
+    "    dir_df['Link to GEO Study'] = dir_df['Signature'].apply(geo_link, clickable=True)\n",
+    "    return dir_df\n",
     "\n",
-    "    return (up_comb_df, dn_comb_df)"
+    "# create download link for table results\n",
+    "def download_link(df, fname):\n",
+    "    df['Link to Bulk RNA-seq Analysis'] = df['Link to Bulk RNA-seq Analysis'].apply(\n",
+    "        lambda x: x.split('href=')[1].split('>')[0].replace('\"', '')\n",
+    "    )\n",
+    "    df['Link to GEO Study'] = df['Link to GEO Study'].apply(\n",
+    "        lambda x: x.split('href=')[1].split('>')[0].replace('\"', '')\n",
+    "    )\n",
+    "    df['Signature'] = df['Signature'].apply(lambda x: x.replace('* ', ''))\n",
+    "    csv = df.to_csv(sep='\\t', index=False)\n",
+    "    b64 = base64.b64encode(csv.encode())\n",
+    "    payload = b64.decode()\n",
+    "    link = f'<a download=\"{fname}\" href=\"data:text/csv;base64,{payload}\" target=\"_blank\">{fname}</a>'\n",
```

Comment:
Please write this to a file instead and link to that file? Inserting tables/images into links is not a good idea in general. Some browsers don't support very large links and will return corrupt files, those links are also not very sharable, etc..